src/services/bunker_cases/abnormal_cl_rebase.py

```diff
@@ -21,187 +22,182 @@
+        curr_lido_balance_with_vault = self._get_validators_balance_with_vault(curr_blockstamp, self.lido_validators)
+        prev_lido_balance_with_vault = self._get_validators_balance_with_vault(prev_blockstamp, prev_lido_validators)
 
-        prev_lido_balance = AbnormalClRebase.calculate_real_balance(prev_lido_validators_by_key)
-        prev_lido_vault_balance = self.w3.lido_contracts.get_withdrawal_balance(prev_blockstamp)
-        prev_lido_balance_with_vault = prev_lido_balance + int(self.w3.from_wei(prev_lido_vault_balance, "gwei"))
+        # Raw CL rebase are calculated as difference between current and previous Lido validators' balances
+        raw_cl_rebase = curr_lido_balance_with_vault - prev_lido_balance_with_vault
 
-        # handle 32 ETH balances of freshly baked validators, who was activated between epochs
-        validators_diff_in_gwei = (len(self.lido_validators) - len(prev_lido_validators_by_key)) * MAX_EFFECTIVE_BALANCE
-        if validators_diff_in_gwei < 0:
-            raise ValueError("Validators count diff should be positive or 0. Something went wrong with CL API")
-
-        corrected_prev_lido_balance_with_vault = (
-            prev_lido_balance_with_vault
-            + validators_diff_in_gwei
-            - self._get_withdrawn_from_vault_between(prev_blockstamp, ref_blockstamp)
+        # We should account validators who have been activated between blocks
+        # And withdrawals from WithdrawalVault
+        validators_count_diff_in_gwei = AbnormalClRebase.calculate_validators_count_diff_in_gwei(
+            self.lido_validators, prev_lido_validators
         )
+        withdrawn_from_vault = self._get_withdrawn_from_vault_between_blocks(prev_blockstamp, curr_blockstamp)
 
-        cl_rebase = ref_lido_balance_with_vault - corrected_prev_lido_balance_with_vault
+        # Finally, we can calculate corrected CL rebase
+        cl_rebase = Gwei(raw_cl_rebase + validators_count_diff_in_gwei + withdrawn_from_vault)
 
-        logger.info(
-            {"msg": f"CL rebase between {prev_blockstamp.ref_epoch,ref_blockstamp.ref_epoch} epochs: {cl_rebase} Gwei"}
-        )
+        logger.info({
+            "msg": f"CL rebase between {prev_blockstamp.ref_epoch,curr_blockstamp.ref_epoch} epochs: {cl_rebase} Gwei"
+        })
 
-        return Gwei(cl_rebase)
+        return cl_rebase
 
-    def _get_withdrawn_from_vault_between(self, prev_blockstamp: ReferenceBlockStamp, curr_blockstamp: ReferenceBlockStamp) -> int:
+    def _get_validators_balance_with_vault(
+        self, blockstamp: ReferenceBlockStamp, lido_validators: list[LidoValidator]
+    ) -> Gwei:
+        """
+        Get Lido validator balance with withdrawals vault balance
+        """
+        real_cl_balance = AbnormalClRebase.calculate_real_balance(lido_validators)
+        withdrawals_vault_balance = int(
+            self.w3.from_wei(self.w3.lido_contracts.get_withdrawal_balance(blockstamp), "gwei")
```

Comment:
`get_withdrawal_balance` has a cache. Shouldn't we use an uncached version here?

Reply:
fixed